src/text/text_formatter.rs

```diff
@@ -417,31 +417,33 @@ impl<'a, W: std::fmt::Write> IonValueFormatter<'a, W> {
         Ok(())
     }
 
-    pub(crate) fn format_sexp(&mut self, value: &SExp) -> IonResult<()> {
+    pub(crate) fn format_sexp<S: AsRef<Sequence>>(&mut self, sequence: S) -> IonResult<()> {
         write!(self.output, "(")?;
-        let mut peekable_itr = value.elements().peekable();
-        while peekable_itr.peek().is_some() {
-            let sexp_value = peekable_itr.next().unwrap();
-            write!(self.output, "{sexp_value}")?;
-            if peekable_itr.peek().is_some() {
-                write!(self.output, " ")?;
-            }
-        }
+        self.format_sequence_elements(sequence, " ")?;
         write!(self.output, ")")?;
         Ok(())
     }
 
-    pub(crate) fn format_list(&mut self, value: &List) -> IonResult<()> {
+    pub(crate) fn format_list<S: AsRef<Sequence>>(&mut self, sequence: S) -> IonResult<()> {
         write!(self.output, "[")?;
-        let mut peekable_itr = value.elements().peekable();
+        self.format_sequence_elements(sequence, ", ")?;
+        write!(self.output, "]")?;
+        Ok(())
+    }
+
+    fn format_sequence_elements<S: AsRef<Sequence>>(
```

Comment:
🗺️ While I was here, I broke out the common sequence formatting functionality into a helper method.